website_field_autocomplete_related/controllers/main.py

```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 LasLabs Inc.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo.http import request
+from odoo.addons.website_field_autocomplete.controllers.main import Website
+
+
+class WebsiteAutocomplete(Website):
+
+    def _get_autocomplete_data(self, model, domain, fields, limit=None):
+        """ Perform recursive search_reads to get all related data """
+        res = super(WebsiteAutocomplete, self)._get_autocomplete_data(
+            model, domain, fields, limit,
+        )
+        self.record_ids = request.env[model].search(domain)
+        for field in fields:
+            if '.' in field:
```

Comment:
Since you have no else clause, you could filter fields and no if would be required